drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c

```diff
@@ -0,0 +1,488 @@
+	if (ret) {
+		return ret;
+	}
+
+	FLEXIO_MCULCD_StartTransfer(flexio_lcd);
+
+	nxp_flexio_lock(config->flexio_dev);
+
+	FLEXIO_MCULCD_WriteCommandBlocking(flexio_lcd, cmd);
+
+	if ((data_buf != NULL) && (len != 0)) {
+		flexio_lcdif_write_data_array(flexio_lcd, data_buf, len);
+	}
+
+	nxp_flexio_unlock(config->flexio_dev);
+
+	FLEXIO_MCULCD_StopTransfer(flexio_lcd);
+
+	return kStatus_Success;
+
+}
+
+static int mipi_dbi_flexio_lcdif_reset(const struct device *dev, uint32_t delay)
+{
+	int err;
+	const struct mcux_flexio_lcdif_config *config = dev->config;
+
+	/* Check if a reset port is provided to reset the LCD controller */
+	if (config->reset.port == NULL) {
+		return 0;
+	}
+
+	/* Reset the LCD controller. */
+	err = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_HIGH);
+	if (err) {
+		return err;
+	}
+
+	err = gpio_pin_set_dt(&config->reset, 0);
+	if (err < 0) {
+		return err;
+	}
+
+	k_usleep(delay);
+
+	err = gpio_pin_set_dt(&config->reset, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	LOG_DBG("%s device reset complete", dev->name);
+
+	return 0;
+}
+
+static int flexio_lcdif_init(const struct device *dev)
+{
+	const struct mcux_flexio_lcdif_config *config = dev->config;
+	struct mcux_flexio_lcdif_data *lcdif_data = dev->data;
+	int err;
+	uint8_t shifter_end = config->child->res.shifter_count - 1;
+
+	if (!device_is_ready(lcdif_data->dma_tx.dma_dev)) {
+		LOG_ERR("%s device is not ready", lcdif_data->dma_tx.dma_dev->name);
+		return -ENODEV;
+	}
+
+	err = nxp_flexio_child_attach(config->flexio_dev, config->child);
+	if (err < 0) {
+		return err;
+	}
+
+	config->flexio_lcd_dev->txShifterStartIndex = config->child->res.shifter_index[0];
+	config->flexio_lcd_dev->txShifterEndIndex = config->child->res.shifter_index[shifter_end];
+
+	config->flexio_lcd_dev->rxShifterStartIndex = config->flexio_lcd_dev->txShifterStartIndex;
+	config->flexio_lcd_dev->rxShifterEndIndex = config->flexio_lcd_dev->txShifterEndIndex;
+
+	config->flexio_lcd_dev->timerIndex = config->child->res.timer_index[0];
+
+	if (config->flexio_lcd_dev->txShifterEndIndex !=
+	    config->flexio_lcd_dev->txShifterStartIndex + shifter_end) {
+		LOG_ERR("Shifters should be continuous");
+		return -ENODEV;
+	}
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err) {
+		return err;
+	}
+
+	/* Pass the FlexIO LCD device as parameter to the function
+	 * callbacks for setting GPIO signals.
+	 */
+	config->flexio_lcd_dev->userData = (void *)dev;
+
+	k_sem_init(&lcdif_data->transfer_done, 0, 1);
+
+	LOG_DBG("%s device init complete", dev->name);
+
+	return 0;
+}
+
+static struct mipi_dbi_driver_api mipi_dbi_lcdif_driver_api = {
+	.reset = mipi_dbi_flexio_lcdif_reset,
+	.command_write = mipi_dbi_flexio_lcdif_command_write,
+	.write_display = mipi_dbi_flexio_ldcif_write_display,
+};
+
+/* The width is read from device tree by the CMake file and passed in as a
+ * compile time define
+ */
+#if (8UL == FLEXIO_MCULCD_DATA_BUS_WIDTH)
```

Comment:
Just want to check here- is there *any* way we can avoid this and still use the SDK driver? Doing this will effectively make the driver single instance.

Reply:
I have deleted this as we have a property for `bus_width` which can be used. However with the compile time define needed for the SDK driver, we still have the limitation of do being able to support multiple instances with different bus widths. I will make a change request to the SDK driver to make this a config property instead of a compile time define.